geo/src/algorithm/sweep/line_or_point.rs

```diff
@@ -186,7 +186,17 @@ impl<T: GeoFloat> LineOrPoint<T> {
             }
         } else {
             line_intersection(self.line(), line).map(|l| match l {
-                LineIntersection::SinglePoint { intersection, .. } => intersection.into(),
+                LineIntersection::SinglePoint { intersection, is_proper } => {
+                    let mut pt = intersection;
+                    if is_proper && (&pt == self.left.deref()) {
+                        if self.left.x == self.right.x {
+                            pt.y = pt.y.next_after(T::infinity());
```

Comment:
The pt is already at the left end of the segment, so nudging towards neg. inf. aggravates the topology issue:  the point will then be completely outside the line-segment as per our ordering.  The asymmetry you notice is because we ignore the corresponding case where the intersection point falls at the right end the segment.  This is because it's a noop that doesn't violate any algo. assumptions / cause panics anyway.  Similarily if pt falls at left of `other` (not `self`)

Reply:
Oh right!  I forgot we already implicitly know their orientation by virtue of the left vs. right, so you're already doing the thing I was asking for.